test-common/test-common-runtime/src/main/java/org/apache/kafka/common/test/ClusterInstance.java

```diff
@@ -258,7 +257,63 @@ default Set<GroupProtocol> supportedGroupProtocols() {
     //---------------------------[wait]---------------------------//
 
     default void waitTopicDeletion(String topic) throws InterruptedException {
-        waitForTopic(topic, 0);
+        Collection<KafkaBroker> brokers = aliveBrokers().values();
+        // wait for metadata
+        TestUtils.waitForCondition(
+            () -> brokers.stream().allMatch(
+                broker -> broker.metadataCache().numPartitions(topic).isEmpty()),
+                60000L, topic + " metadata not propagated after 60000 ms");
+
+        for (ControllerServer controller : controllers().values()) {
+            long controllerOffset = controller.raftManager().replicatedLog().endOffset().offset() - 1;
+            TestUtils.waitForCondition(
+                () -> brokers.stream().allMatch(broker -> ((BrokerServer) broker).sharedServer().loader().lastAppliedOffset() >= controllerOffset),
+                60000L, "Timeout waiting for controller metadata propagating to brokers");
+        }
+
+        List<TopicPartition> topicPartitions = List.of(new TopicPartition(topic, 0));
```

Comment:
we don't need to use array, right?

Reply:
Yes, we don't need it. It's odd that the original logic uses intStream to create a list, but we could clean up in this PR.